### PR TITLE
Filter capabilities and materials search

### PR DESCRIPTION
**Pull request by DiegoAstiazaran**
This PR addresses issue #139 :
- A method was added to the home controller, capabilities and materials. These return a list of items that can be filtered if a capability or material (the one not being queried) is passed and if a string is passed to match the name of the items returned.
- The defaultPillBox container was removed and only one container is used now.
   - It is filled with all the capabilities and materials when the page first loads, only one of them if that type is being queried in the input text area. Also all of them if one has already been selected for each type.
- The selected pill is now aligned in the middle of its container
- The input text for the searches is now aligned in the middle.

### CODE REVIEW

File: app/controllers/home_controller.rb [hunk 1]
```diff
@@ -1,16 +1,67 @@
...
+      @capabilities = Capability.select(:id, :name).where(id: capabilities)
+    else
+      # Get all the capabilities
+      @capabilities = Capability.select(:id, :name)
+    end
+    if params[:query].present?
+      # Filter capabilities by query written
+      @capabilities = @capabilities.select(:id, :name).where("name ILIKE ?", "%#{params[:query]}%").as_json()
+    end
+    if request.xhr?
```
Comment (erik0686) [inline app/controllers/home_controller.rb]: Could some of this lines in both methods be factored out?? They are basically repeated methods, but you might be able to better answer this

File: app/views/home/landing.html.erb [hunk 2]
```diff
@@ -22,121 +22,173 @@
...
-          $defaultPillBox.hide();
-          $pillBox.show();
-          draw(results, collection);
-        };
-      };
+      // Creates the pills for the items in results (array)
+      function draw(results, type) {
+        let pillsHTML = getHTMLForPills(results, type);
+        $pillBox.html(pillsHTML);
+      }
```
Comment (erik0686) [inline app/views/home/landing.html.erb]: You're changing most of the js, would it be better to make a separate js file?

File: app/controllers/home_controller.rb [hunk 3]
```diff
@@ -1,16 +1,61 @@
...
+      @capabilities = Capability.select(:id, :name).where(id: capabilities)
+    else
+      # Get all the capabilities
+      @capabilities = Capability.select(:id, :name)
+    end
+    query = params[:query]
+    if query.present?
+      # Filter capabilities by query written
+      @capabilities = @capabilities.where('name ILIKE ?', "%#{query}%").as_json
+    end
```
Comment (erik0686) [inline app/controllers/home_controller.rb]: This two methods are basically the same, could some things be factored out?

File: app/controllers/home_controller.rb (hunk 1 above)
Comment (DiegoAstiazaran) [reply app/controllers/home_controller.rb]: Done :smiley:

File: app/views/home/landing.html.erb (hunk 2 above)
Comment (DiegoAstiazaran) [reply app/views/home/landing.html.erb]: Done

**Pull request merged**